sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeScreen.kt

```diff
@@ -19,26 +19,34 @@ import co.nimblehq.sample.compose.ui.screens.AppBar
 import co.nimblehq.sample.compose.ui.theme.ComposeTheme
 import co.nimblehq.sample.compose.ui.userReadableMessage
 import com.google.accompanist.permissions.*
+import kotlinx.coroutines.delay
 
 @Composable
 fun HomeScreen(
     viewModel: HomeViewModel = hiltViewModel(),
     navigator: (destination: AppDestination) -> Unit
 ) {
+    val showLoading: IsLoading by viewModel.showLoading.collectAsState()
+    val uiModels: List<UiModel> by viewModel.uiModels.collectAsState()
+    val firstTimeLaunch: Boolean by viewModel.firstTimeLaunch.collectAsState()
+
     val context = LocalContext.current
     LaunchedEffect(viewModel.error) {
         viewModel.error.collect { error ->
             val message = error.userReadableMessage(context)
             Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
         }
     }
+    LaunchedEffect(firstTimeLaunch) {
+        delay(1000)
```

Comment:
@lydiasama I know that it is only the sample, but could we put the delay in `ViewModel`? I'm not feeling good when we block the UI thread 🤔

Reply:
@doannimble The point is we have too many toast messages to show here. We cannot guarantee when it emits the `firstTimeLanch` stream here because it's all asynchronous. Does it make more sense to put this delay on the `ViewModel`?

Reply:
Not so sure about this delay here @lydiasama 🤔 When I tested it on my device without the `delay(1000)` it works fine and can show the Toast `This is the first time launch` normally 😄 

https://user-images.githubusercontent.com/25218255/216553501-2a08c739-2c7b-44f6-8920-4a093d845678.mp4